helpers/cleanup.sh

```diff
@@ -13,8 +13,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+# Usage info
+show_help() {
+cat << EOF
+Usage: ${0##*/} [-p PROJECT_ID] [-f HOST_PROJECT] [-o ORG_ID]
+Remove service account created by setup.sh script.
+
+    -p PROJECT_ID         project id where service account will be created.
+    -s SERVICE ACCOUNT    service account name.
+    -f HOST_PROJECT       id of a project holding shared vpc.
+    -o ORG_ID             organization id.
+    -h                    this help.
+Example: ./cleanup.sh -p test-project-1 -s cloud-foundation-forseti -o organization.com
+EOF
+}
+
+# Initialize opt variables:
+project_id=""
+host_project=""
+service_account_id=""
+
+OPTIND=1
+while getopts ":h:p:s:f:o:" opt; do
+    case $opt in
+        h)
+            show_help
+            exit 0
+            ;;
+        p)  project_id=$OPTARG
+            ;;
+        s)  service_account_id=$OPTARG
+            ;;
+        f)  host_project=$OPTARG
+            ;;
+        *)
+            show_help >&2
+            exit 1
+            ;;
+    esac
+done
+shift "$((OPTIND-1))"   # Discard the options and sentinel --
+
+if [[ $project_id == "" ]];
+then
+    echo "ERROR -p option is mandatory"
```

Comment:
An option being mandatory is a contradiction. Mandatory arguments should be positional.

Reply:
Argument parsing styles are pretty subjective; I've seen (and written) a number of CLI tools that have used both positional arguments and required flags. Requiring flags for values is slightly more verbose but increases readability, which seems like a decent tradeoff.

In addition argument parsing in bash is generally less than pleasant, and mixing positional arguments with flags is generally a nightmare. For the sake of simplicity and readability, I actually prefer the way that we have this now. Is there a cost to this approach that I'm not seeing?